src/GraphQL/Builders/ConnectionBuilder_Typed.cs

```diff
@@ -0,0 +1,291 @@
+        /// Adds an argument to the connection field.
+        /// </summary>
+        /// <typeparam name="TArgumentGraphType">The graph type of the argument.</typeparam>
+        /// <typeparam name="TArgumentType">The type of the argument value.</typeparam>
+        /// <param name="name">The name of the argument.</param>
+        /// <param name="description">The description of the argument.</param>
+        /// <param name="defaultValue">The default value of the argument.</param>
+        /// <param name="configure">A delegate to further configure the argument.</param>
+        public ConnectionBuilder<TSourceType, TReturnType> Argument<TArgumentGraphType, TArgumentType>(string name, string? description,
+#if NETSTANDARD2_1
+            [AllowNull]
+#endif
+            TArgumentType defaultValue = default!, Action<QueryArgument>? configure = null)
+            where TArgumentGraphType : IGraphType
+            => Argument<TArgumentGraphType>(name, arg =>
+            {
+                arg.Description = description;
+                arg.DefaultValue = defaultValue;
+                configure?.Invoke(arg);
+            });
+
+        /// <summary>
+        /// Runs a configuration delegate for the connection field.
+        /// </summary>
+        public virtual ConnectionBuilder<TSourceType, TReturnType> Configure(Action<FieldType> configure)
+        {
+            configure(FieldType);
+            return this;
+        }
+
+        /// <summary>
+        /// Apply directive to connection field without specifying arguments. If the directive
+        /// declaration has arguments, then their default values (if any) will be used.
+        /// </summary>
+        /// <param name="name">Directive name.</param>
+        public virtual ConnectionBuilder<TSourceType, TReturnType> Directive(string name)
+        {
+            FieldType.ApplyDirective(name);
+            return this;
+        }
+
+        /// <summary>
+        /// Apply directive to connection field specifying one argument. If the directive
+        /// declaration has other arguments, then their default values (if any) will be used.
+        /// </summary>
+        /// <param name="name">Directive name.</param>
+        /// <param name="argumentName">Argument name.</param>
+        /// <param name="argumentValue">Argument value.</param>
+        public virtual ConnectionBuilder<TSourceType, TReturnType> Directive(string name, string argumentName, object? argumentValue)
+        {
+            FieldType.ApplyDirective(name, argumentName, argumentValue);
+            return this;
+        }
+
+        /// <summary>
+        /// Apply directive to connection field specifying configuration delegate.
+        /// </summary>
+        /// <param name="name">Directive name.</param>
+        /// <param name="configure">Configuration delegate.</param>
+        public virtual ConnectionBuilder<TSourceType, TReturnType> Directive(string name, Action<AppliedDirective> configure)
```

Comment:
Why `Configure` and three `Directive` methods are `virtual`?

Reply:
I copied and pasted your code suggestions from the other PR and you had written them as virtual. I thought I had removed virtual but apparently I didn’t. I can fix it.

Reply:
Ooops. I remember! I did them `virtual` because all other methods in `FieldBuilder` are `virtual` too.

Reply:
See #1351. So I would to do all methods in `ConnectionBuilder` as `virtual`.